src/pipx/backend.py

```diff
@@ -0,0 +1,21 @@
+import logging
+import shutil
+
+SUPPORTED_VENV_BACKENDS = ("uv", "venv", "virtualenv")
+SUPPORTED_INSTALLERS = ("uv", "pip")
+
+logger = logging.getLogger(__name__)
+
+
+def path_to_exec(executable: str, installer: bool = False) -> str:
+    if executable == "venv" or "pip":
+        return executable
+    path = shutil.which(executable)
+    if path:
+        return path
+    elif installer:
+        logger.warning(f"{executable} not found on PATH. Falling back to pip.")
+        return ""
+    else:
+        logger.warning(f"{executable} not found on PATH. Falling back to venv.")
+        return ""
```

Comment:
We might as well return the program names here, to cut down on conditions below?

```suggestion
    elif installer:
        logger.warning(f"'{executable}' not found on PATH. Falling back to 'pip'.")
        return "pip"
    else:
        logger.warning(f"'{executable}' not found on PATH. Falling back to 'venv'.")
        return "venv"
```

Reply:
The empty string is used to determine the backend and installer to use here:
```py
        if not path_to_exec(self.backend):
            self.backend = "venv"
        if not path_to_exec(self.installer, installer=True):
            self.installer = "pip"
```

Reply:
I know, I thought we could assign directly:

```python
self.backend = path_to_exec(self.backend)
self.installer = path_to_exec(self.installer, installer=True)
```

But I guess that's not doable, as in case of `uv` or `virtualenv`  a path would be returned.

Reply:
> But I guess that's not doable, as in case of uv or virtualenv a path would be returned.

Yes exactly.